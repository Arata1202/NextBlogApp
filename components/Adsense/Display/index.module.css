.unit {
  max-width: 100%;
  min-height: 280px;
}

.ins {
  display: flex;
  justify-content: center;
  width: 100%;
}

@media (max-width: 640px) {
  .unit {
    min-height: 200px;
  }
}
