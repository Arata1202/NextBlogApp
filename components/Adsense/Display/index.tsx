'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';
import { useTheme } from 'next-themes';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type DisplayProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const Display = ({ slot, format = 'rectangle', responsive = 'false', style }: DisplayProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  const { theme } = useTheme();

  return (
    <div
      style={{ maxWidth: '100%' }}
      className="FirstAd"
      key={pathname.replace(/\//g, '-') + '-' + slot}
    >
      <p className={`text-center ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}>
        スポンサーリンク
      </p>
      <ins
        className="adsbygoogle"
        style={{
          display: 'flex',
          justifyContent: 'center',
          width: '100%',
          height: '250px',
          ...style,
        }}
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default Display;
