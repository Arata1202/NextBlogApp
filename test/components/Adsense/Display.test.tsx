import { render } from '@testing-library/react';
import Display from '@/components/Adsense/Display';

describe('Display', () => {
  test('スナップショット（Display)', () => {
    const { asFragment } = render(<Display slot="00000" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
