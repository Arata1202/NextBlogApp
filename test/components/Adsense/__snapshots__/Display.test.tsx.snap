// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display スナップショット（Display) 1`] = `
<DocumentFragment>
  <div
    class="FirstAd"
    style="max-width: 100%; overflow: hidden;"
  >
    <p
      class="text-center"
    >
      スポンサーリンク
    </p>
    <ins
      class="adsbygoogle"
      data-ad-client="ca-pub-1705865999592590"
      data-ad-format="rectangle"
      data-ad-slot="00000"
      data-full-width-responsive="false"
      style="display: flex; justify-content: center; width: 100%;"
    />
  </div>
</DocumentFragment>
`;
